### ISSUE TITLE
Cannot use **default** as argument

### ISSUE THREAD

**Issue by harshadyeola** [datafolklabs/cement#294]
Hi There..
Cannot use **default** as argument!!
```python
class MyappDeleteController(CementBaseController):
    class Meta:
        label = 'delete'
        stacked_on = 'site'
        stacked_type = 'nested'
        description = 'delete an existing website'
        arguments = [
            (['arg1'],
                dict(help='domain name to be deleted')),
            (['--no-prompt'],
                dict(help="doesnt ask permission for delete",
            ]
    @expose(hide=True)
    def default(self):
        Log.info(self, "in default")
        # Delete code here
```
With above code i have used default as argument, but its not taking default as argument but as command.
```
root@ubuntu-web:~# myapp site delete default 
usage: myapp (sub-commands ...) [options ...] {arguments ...}
myapp: error: the following arguments are required: arg1
```

**Comment by derks** [comment]
This is an interesting situation, because the `default` command is executed if no sub-command is passed, however in your use case you need to pass an argument of `default` (which in turn is read as explicitly calling the `default` sub-command).

I would suggest trying the following.  Currently, you are stacking this controller on top of the `site` controller as `nested`.  Meaning you would call this as:

```
$ yourapp site delete [arg1]
```

Would it be possible to instead make this an `embedded` controller, with a `delete` sub-command?  For example:

```python
class MyappDeleteController(CementBaseController):
    class Meta:
        label = 'delete'
        stacked_on = 'site'
        stacked_type = 'embedded'
        description = 'delete an existing website'
        arguments = [
            (['arg1'],
                dict(help='domain name to be deleted')),
            (['--no-prompt'],
                dict(help="doesnt ask permission for delete",
            ]
    @expose()
    def delete(self):
        Log.info(self, "in delete")
        # Delete code here
```

The `delete` sub-command and arguments would then be available under the `site` namespace.  That is one option.  Alternatively, you could make `arg1` optional, and then handle the requirement in your code.  This is kind of messy, but it could work... basically, check that `default` was explicitly passed via command line:

```
import sys

class MyappDeleteController(CementBaseController):
    class Meta:
        label = 'delete'
        stacked_on = 'site'
        stacked_type = 'nested'
        description = 'delete an existing website'
        arguments = [
            (['arg1'],
                dict(help='domain name to be deleted', nargs='?')),
            (['--no-prompt'],
                dict(help="doesnt ask permission for delete",
            ]
    
    @expose(hide=True)
    def default(self):
        try:
            # change the sys.argv index number to match your app layout
            if sys.argv[2] == 'default':
                self.app.pargs.arg1 = 'default'
        except IndexError as e:
            # not that many arguments were passed
            pass

        # continue normal operations
        assert self.app.pargs.arg1, "Argument required"

        print "Inside MySecondController.default()"
        print "Arg1 => %s" % self.app.pargs.arg1
```

In the above example.. the following would throw an `Argument required` error:

```
$ yourapp site delete
```

But this would give the end result of `arg1` producing `default`:

```
$ yourapp site delete default
```


The important thing to note in the above is the use of `nargs='?'` on the `arg1` argument... which makes it optional, otherwise it would throw a 'missing arguments' error.  

I know that is probably confusing... and not ideal, but does that help at all?

**Comment by harshadyeola** [comment]
@derks 
Thanks for quick solution, I will try this this and back to you soon.

**Comment by harshadyeola** [comment]
@derks 
I can't use  ``` MyappDeleteController``` with ```stacked_type``` as ```embedded```.  as i have different optional arguments.

**Comment by derks** [comment]
@harshadyeola I am sorry I haven't not gotten back to this.  Have you found any alternative solutions?

**Comment by derks** [comment]
I've committed a work-around for this issue to cement/master, which will be available in Cement 2.6 Stable.  The following is a working example:

```python
from cement.core import handler
from cement.core.foundation import CementApp
from cement.core.controller import CementBaseController, expose

class BaseController(CementBaseController):
    class Meta:
        label = 'base'

    @expose(hide=True)
    def default(self):
        print('Inside MyController.default()')

class DeleteController(CementBaseController):
    class Meta:
        label = 'delete'
        stacked_on = 'base'
        stacked_type = 'nested'
        arguments = [
            (['arg1'],
             dict(help='domain name to be deleted', nargs='?')),
            ]
        default_func = 'default_command'

    @expose(hide=True)
    def default_command(self):
        print(self.app.argv)
        print('Inside DeleteController.default_command()')
        print('arg1 => %s' % self.app.pargs.arg1)


class MyApp(CementApp):
    class Meta:
        label = 'myapp'
        base_controller = BaseController

with MyApp() as app:
    handler.register(DeleteController)

    try:
        app.run()
    except Exception as e:
        print e
```

In action:

```
$ python myapp.py delete default
Inside DeleteController.default_command()
arg1 => default
```


The above uses the new `CementBaseController.Meta.default_func` meta-data option to change the default function name.  Additionally, the `default()` function is renamed to `default_command` but it can be called anything (can **not** start with an `_` however).

Please let me know if this doesn't work for you.

**Comment by harshadyeola** [comment]
@derks 
Thanks for the heads up. I will try your solution and let you know.
